### DISCUSSION TITLE
YouTube video without functional file links.

### DISCUSSION THREAD

**Discussion by MTCMTC000** [aclap-dev/video-downloadhelper#28] [Help and Q&A]
### Your question

I am on:
Video DownloadHelper
Version 8.2.0.18a1 (beta)
Browser locale: en-US
Production build
Built on 2024-02-01
Build options: browser=firefox, linuxlic=false, noyt=false, target=mozilla

Platform Mac aarch64
Browser Mozilla Firefox 122.0

VdhCoApp 2.0.10

This video has only one download option appear and it fail.  I am not having any problems getting a list of options for all other YouTube videos.
https://www.youtube.com/watch?v=5Re2r2i1jpQ

It took me a few minutes to get through the shock of the latest updates but I have quickly grown to like them.  I am getting faster downloads from YouTube.  However, I am getting some fails.  When the fails come it generally just shows a "Native Port Disconnect" without any naming of the video that failed.  I have no way to know what video page to reopen to try again.  I can easily create a list of 50 videos to download so I have to go back and sort through the browser history to check titles that are downloaded.  Since they download order is now random, I cannot coordinate lists to try to find it.  Note, I have actually limited the downloads to one.  The download rate is taking all my bandwidth so I have no benefit of downloading more than one.

**Comment by MTCMTC000** [comment]
UPDATE.  The video options are now available.  I will make a second post about the other issue if it continues.

**Comment by GrampaWildWilly** [comment]
Apparently I'm a bit slow.  I was preparing this post when you posted.  Oh well.  I don't want this to go to waste so . . .

You are on the latest Firefox & the latest beta of VDH but you're not on the latest CoApp.  Seems a bit odd.  Here's what I'm running: Windows 11 Pro 64-bit version 23H2 build 22631.3085, Firefox 122.0 64-bit, licensed VDH 8.2.0.18a1 beta, CoApp 2.0.14.  You really ought to get the current CoApp.

I went to your video & got this, which I gather you are also now getting:

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/3539171e-fc60-43c4-ac1b-b64e1e12ce63)

Full disclosure.  After downloading the mp4, when I clicked the mkv, VDH told me the file no longer existed.  It showed a 404 Not Found error.  Strange.  That seems like a YouTube issue, not a VDH issue.  So I hit F5 to reload the YouTube page.  That added a duplicate set of variants to the VDH menu.  The mkv in the duplicate set downloaded just fine.  Each download took between 2 & 3 seconds.  Each one played fine in VLC.

I will admit that Native Port Disconnect is not an error I've ever encountered.  I'm not totally sure what it means, either.  But it sounds vaguely to me like some sort of Internet connectivity issue.  Meaning, the problem lies somewhere in the area of your machine, your operating system, & your Internet connection.  VDH seems like more the victim than the culprit on this one.  If you find it's happening intermittently, it could just be your ISP telling you you're trying to pump too much through your Internet connection.  When it comes to concurrent downloads, there does come a point of diminishing returns.  I'm finding that after I've got 3 or 4 downloads going at the same time, all of them screaming fast, any further downloads I pile on are not going to go too fast, plus they will slow down the ones that were already there.  This is a function of the bandwidth you are buying from your ISP.  You will have to determine the point of no return for your setup.  In Windows, we have the Resource Monitor that lets me see the speed of my downloads.  I have to assume there is a similar tool on the Mac.  You should monitor what that tool tells you over a span of time to get a feel for just how much you can get away with in the way of concurrent downloads.

Just so you can see what I'm talking about, I put a screenshot of the Resource Monitor in this discussion:

https://github.com/aclap-dev/video-downloadhelper/discussions/20

**Comment by MTCMTC000** [comment]
I checked the CoApp when I was copying the system info.  I click "ReCheck" but it did not come back with an option to update.  I learned from CAD software to not update unless the system was wanting the update.  Otherwise, it could install software that does not fit your system.  Now that I know the built-in updater does not get it right, maybe a beta version issue, I will manually check it.

The image you posted is correct.  I was able to download it without problem.  

I normally just download the mp4 files.  I had tried the mkv files and had the issue but went no further trying to make it work.  I have had the limited menu issue appear on several videos but when I the file completely load, I'm on a slow connect, the menu would fully populate or a refresh would let it populate.  I agree that VDH is not the core problem.  If I am using my phone for internet access, I can have various other unrepeatable bugs.  The real reason I sent the original report was because repeated refresh cycles would not allow the menu to populate.  I do monitor the download graph.  The "Native Port Disconnect" has been happening on the phone and the WiFi connect I to which have access.

Thank you for the reply.  I hope all is well with you.

**Comment by GrampaWildWilly** [comment]
Sounds like this problem is actually resolved.  You should close it.  I put instructions for that here:

https://github.com/aclap-dev/video-downloadhelper/discussions/24
